NEMO/interlocks.py

```diff
@@ -201,6 +201,44 @@ def _send_command(self, interlock: Interlock_model, command_type: Interlock_mode
 			sock.close()
 
 
+class ProXrInterlock(Interlock):
+	"""Very basic support for ProXR relay controllers.
+	Supports on/off for all relays in bank 1.
+	See https://ncd.io/proxr-quick-start-guide/ for more about ProXR.
+	"""
+	# supported proxr commands
+	PXR_CMD_B1_OFF = (254, 129, 1)
+	PXR_CMD_B1_ON = (254, 130, 1)
+
+	def _send_bytes(self, relay, proxrcmd):
+		"""Returns the response from the relay controller.
+		Argument relay is a connected socket object.
+		Argument proxrcmd (the ProXR command) is an iterable of 8-bit integers.
+		"""
+		# make sure that all bytes get sent
+		bytes_sent = 0
+		while bytes_sent < len(proxrcmd):
+			bytes_sent += relay.send(bytes(proxrcmd[bytes_sent:]))
+		# relay responses can include erroneous data
+		# only the last byte of the response is important
+		return relay.recv(64)[-1]
+
+	def _send_command(self, interlock: Interlock_model, command_type: Interlock_model.State) -> Interlock_model.State:
+		"""Returns and sets NEMO locked/unlocked state."""
+		state = Interlock_model.State.UNKNOWN
+		try:
+			with socket.create_connection((interlock.card.server, interlock.card.port), 10) as relay:
+				if command_type == Interlock_model.State.LOCKED:
+					self._send_bytes(relay, self.PXR_CMD_B1_OFF)
+					state = Interlock_model.State.LOCKED
+				elif command_type == Interlock_model.State.UNLOCKED:
+					self._send_bytes(relay, self.PXR_CMD_B1_ON)
+					state = Interlock_model.State.UNLOCKED
```

Comment:
now if sending the data is successful, even if there is an error in the controller itself, the state will be set to locked/unlocked even though it might not be. (this comment follows the other above about the result of the command)
I think it would make more sense to return the result of a stat command. this way, the result will be what the relay state really is.